src/transformers/models/perceiver/modeling_perceiver.py

```diff
@@ -2850,14 +2850,14 @@ def __init__(self, config: PerceiverConfig) -> None:
     def num_channels(self) -> int:
         return self.config.d_model
 
-    def forward(self, inputs: torch.LongTensor) -> torch.FloatTensor:
+    def forward(self, inputs: torch.LongTensor, pos: Optional[torch.Tensor] = None, network_input_is_1d: bool = True):
         embeddings = self.embeddings(inputs)
 
         seq_length = inputs.shape[1]
         position_ids = torch.arange(0, seq_length, device=inputs.device)
         embeddings = embeddings + self.position_embeddings(position_ids)
 
-        return embeddings, None, None
+        return embeddings, None, embeddings
```

Comment:
Following the other modalities, I believe the final returned object should be the embeddings without the positional embeddings i.e. `embeddings_without_pos`.

Reply:
Good point! I pushed a fix.

Thanks for reviewing by the way :-)

Reply:
My pleasure :)